glotaran/builtin/megacomplexes/decay/util.py

```diff
@@ -271,42 +297,45 @@ def retrieve_decay_associated_data(
     matrix = k_matrix.full(species)
     matrix_reduced = k_matrix.reduced(species)
     a_matrix = megacomplex.get_a_matrix(dataset_model)
-    rates = k_matrix.rates(species, initial_concentration)
+    rates = -1 * k_matrix.rates(species, initial_concentration)
```

Comment:
This appears of the category, oh the answers is the wrong sign, so let's multiple it with another factor of -1 somewhere. But two wrongs don't make a right. I would expect a check somewhere to make sure the lifetime is positive when it should be, and that the rates that represent decays are of the appropriate sign when they should.

In any case why is it needed that rates is multiplied with -1, it should have been the correct sign already, and the multiplication should happen in the calculation of lifetimes. And even then it might make more sense to have a helper function that ensures correct sign, and throws a warning when something is wrong unexpectedly. 

In any case, all unit tests are passing so we have probably multiplied the correct number of times with -1, but I'd recommend a future refactor that solves this more elegantly and adds some unit tests.

Reply:
Actually, let us please create an issue motivating why the change is being made, and why before it was not as it should be, and now it is as it should be.

Reply:
By definition, the eigenvalues of the compartmental model are negative, and previously these eigenvalues were then reported as the rates. however, the rates are the negatives of the eigenvalues, thus this multiplication with -1 was overdue. then the lifetimes which are the reciprocals of the rates are also positive.  I recommend to add a comment line # the rates are the negatives of the eigenvalues of the compartmental model

Reply:
@ism200 I think the point @jsnel wants to make is that `k_matrix.rates` should already return the proper rates instead of multiplying them with -1 right before we save them.
Current implementation:
https://github.com/glotaran/pyglotaran/blob/a635bb1dea965777fab77f70f536ae3c269c2e84/glotaran/builtin/megacomplexes/decay/k_matrix.py#L208-L219

Reply:
in that case, you can put 
rates, _ = -self.eigen(compartments)
and add before line 218 a comment line # the rates are the negatives of the eigenvalues of the compartmental model

Reply:
This should be fixed now ([ceed5cb](https://github.com/glotaran/pyglotaran/pull/1022/commits/ceed5cb34ca2faea0054c1cb820b9ff0c5792b47)-[40f6e75](https://github.com/glotaran/pyglotaran/pull/1022/commits/40f6e7526c55088a46f71b5d4f9be057f2727583))